packages/terminal/src/browser/terminal-frontend-contribution.ts

```diff
@@ -54,14 +69,91 @@ export class TerminalFrontendContribution implements CommandContribution, MenuCo
     }
 
     registerKeybindings(keybindings: KeybindingRegistry): void {
-        [
-            {
-                commandId: TerminalCommands.NEW.id,
-                keyCode: KeyCode.createKeyCode({ first: Key.BACKQUOTE, modifiers: [Modifier.M1] })
-            },
-        ].forEach(binding => {
-            keybindings.registerKeybinding(binding);
+        this.keybindingContextRegistry.registerContext({
+            id: TERMINAL_ACTIVE_CONTEXT,
+            isEnabled: isTerminalFocused,
         });
+
+        keybindings.registerKeybindings({
+            commandId: TerminalCommands.NEW.id,
+            keyCode: KeyCode.createKeyCode({ first: Key.BACKQUOTE, modifiers: [Modifier.M1] }),
+        });
+
+        /* Register passthrough keybindings for combinations recognized by
+           xterm.js and converted to control characters.
+
+             See: https://github.com/xtermjs/xterm.js/blob/v3/src/Terminal.ts#L1684 */
+
+        /* Register ctrl + k (the passed Key) as a passthrough command in the
+           context of the terminal.  */
+        const regCtrl = (k: Key) => {
+            keybindings.registerKeybindings({
+                commandId: KeybindingRegistry.PASSTHROUGH_PSEUDO_COMMAND,
+                keyCode: KeyCode.createKeyCode({ first: k, modifiers: [Modifier.M1] }),
+                context: this.keybindingContextRegistry.getContext(TERMINAL_ACTIVE_CONTEXT),
```

Comment:
Since it has not been used so far, could we change this property to type string. I.e. you assign the id?

Reply:
Yes, it will be simpler as well.